### PR TITLE
Distro command fix for #1145

### PR DESCRIPTION
**Pull request by gamerson**

### CODE REVIEW

File: biz.aQute.bnd/bnd.bnd [hunk 1]
```diff
@@ -33,6 +33,7 @@ Bundle-Description: This command line utility is the Swiss army knife of OSGi. I
 -buildpath: \
 	aQute.libg;version=latest,\
 	biz.aQute.bndlib;version=project,\
+	org.apache.felix.framework;version=latest;packages=*,\
```
Comment (bjhargrave) [inline biz.aQute.bnd/bnd.bnd]: This seems very wrong. Why is this needed on the `-buildpath`? I see you use it in the test class, perhaps you meant to put it on the `-testpath`?

File: biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	@Override
+	protected void setUp() throws Exception {
+		tmp = IO.getFile("generated/tmp");
+		tmp.mkdirs();
+
+		configuration = new HashMap<String,Object>();
+		configuration.put(Constants.FRAMEWORK_STORAGE_CLEAN, Constants.FRAMEWORK_STORAGE_CLEAN_ONFIRSTINIT);
+		configuration.put(Constants.FRAMEWORK_STORAGE, new File(tmp, "fwstorage").getAbsolutePath());
+		configuration.put(Constants.FRAMEWORK_SYSTEMPACKAGES_EXTRA, "org.osgi.framework.launch;version=1.4");
+		framework = new org.apache.felix.framework.FrameworkFactory().newFramework(configuration);
```
Comment (bjhargrave) [inline biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java]: Why aren't use using a normal launcher which uses ServiceLoader to find the FrameworkFactory object?

File: biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		framework.init();
+		framework.start();
+		BundleContext context = framework.getBundleContext();
+
+		String[] bundles = {
+				"../cnf/repo/osgi.cmpn/osgi.cmpn-4.3.1.jar", "../biz.aQute.remote/generated/biz.aQute.remote.agent.jar",
```
Comment (bjhargrave) [inline biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java]: osgi.cmpn jars are not meant to be uses as an installed bundle. Why is this needed?

File: biz.aQute.bnd/bnd.bnd (hunk 1 above)
Comment (gamerson) [inline biz.aQute.bnd/bnd.bnd]: Ah yes -testpath is all that is needed.

File: biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java (hunk 2 above)
Comment (gamerson) [inline biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java]: I simply copied this from another test class that used the same thing to setup a framework for test, I can replace it with Launcher

File: biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java (hunk 3 above)
Comment (gamerson) [inline biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java]: not needed, I can remove it.

File: biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java (hunk 2 above)
Comment (gamerson) [inline biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java]: So on second thought I will need some help with this.  Can you point me to an example of where the Launcher is used to start the framework instead of just calling it directly?  thanks.

Comment (bjhargrave) [inline biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java]: Well the launcher starts a separate process. See https://github.com/bndtools/bnd/blob/master/biz.aQute.remote/src/aQute/remote/agent/AgentDispatcher.java#L75 for using ServiceLoader to start a framework in the current process.

File: biz.aQute.bnd/bnd.bnd [hunk 4]
```diff
@@ -44,7 +44,10 @@ Bundle-Description: This command line utility is the Swiss army knife of OSGi. I
 	biz.aQute.remote.api;version=latest,\
 	org.yaml.snakeyaml;version=1.15
 
--testpath: ${junit}, biz.aQute.bnd.embedded-repo; version=snapshot
+-testpath: \
+	${junit},\
+	biz.aQute.bnd.embedded-repo; version=snapshot,\
+	org.apache.felix.framework;version=latest;packages=*
```
Comment (bjhargrave) [inline biz.aQute.bnd/bnd.bnd]: Now that you have removed the direct reference to Felix code (by using ServiceLoader), you should remove `packages=*` from the `-testpath` entry.

**Comment by bjhargrave** [comment]
Would it be possible to squash these commits down into a smaller number? Some of the commits are fixes to earlier commits based upon feedback.

**Comment by gamerson** [comment]
Updated and squashed.

**Comment by bjhargrave** [comment]
LGTM

**Comment by gamerson** [comment]
Thanks for the review @bjhargrave
